game-core/src/main/java/games/strategy/triplea/delegate/battle/casualty/CasualtySortingUtil.java

```diff
@@ -10,24 +11,18 @@
 
 @UtilityClass
 public class CasualtySortingUtil {
-  /** In an amphibious assault, sort marines with higher scores last */
   public static void sortPreBattle(final List<Unit> units) {
     units.sort(
         Comparator.comparing(Unit::getType, Comparator.comparing(UnitType::getName))
-            .thenComparing(compareMarines())
+            .thenComparing(CasualtySortingUtil::compareAmphibiousUnits)
             .thenComparing(UnitComparator.getLowestToHighestMovementComparator()));
   }
 
-  public static Comparator<Unit> compareMarines() {
-    return (u1, u2) -> {
-      final int result = Boolean.compare(u1.getWasAmphibious(), u2.getWasAmphibious());
-      if (result != 0) {
-        return result;
-      }
-
-      final UnitAttachment ua1 = UnitAttachment.get(u1.getType());
-      final UnitAttachment ua2 = UnitAttachment.get(u2.getType());
-      return Integer.compare(ua1.getIsMarine(), ua2.getIsMarine());
```

Comment:
Problem is here, the fact that we are comparing marines or amphib is red herring and wrong. The fix is not improving this IMO, but changing this to be able to compute unit power.

Reply:
If we can then go on to do a secondary sort, we'll be fixing a second bug where planes with less movement are selected after units with more movement.

Reply:
It already has a secondary sort by movement.  So if that bug exists, then this sort won't fix that bug.